app/src/main/java/com/lastaoutdoor/lasta/ui/screen/settings/SettingsComponent.kt

```diff
@@ -31,131 +42,221 @@ fun SettingsComponent(
     updateClimbingLevel: (UserLevel) -> Unit,
     updateHikingLevel: (UserLevel) -> Unit,
     updateBikingLevel: (UserLevel) -> Unit,
+    setUpOrSetting: String // "Setup" or "Settings"
 ) {
-  var selectedLanguage = language
-  var selectedActivity = prefActivity
+  Column(horizontalAlignment = Alignment.CenterHorizontally) {
+    TitleComponent(setUpOrSetting = setUpOrSetting) // Title of the screen
+    Spacer(modifier = Modifier.height(24.dp))
+    SettingsHeader(type = "General") // Header for General Settings
+    Spacer(modifier = Modifier.height(24.dp))
+    LanguageSelectionComponent(language, updateLanguage)
+    Spacer(modifier = Modifier.height(24.dp))
+    SettingsHeader(type = "Activity") // Header for Activity settings
+    Spacer(modifier = Modifier.height(24.dp))
+    FavoriteActivityComponent(prefActivity, updatePrefActivity) // Favorite Activity Selection
+    Spacer(modifier = Modifier.height(24.dp))
+    ActivityLevelsComponent(levels, updateClimbingLevel, updateHikingLevel, updateBikingLevel)
+  }
+}
 
-  var climbingLevel = levels.climbingLevel
-  var hikingLevel = levels.hikingLevel
-  var bikingLevel = levels.bikingLevel
+@Composable
+fun SettingsHeader(type: String) {
```

Comment:
There exists an `ActivityType` enum, you should use it instead of a `String` parameter to avoid faulty usage. Ex : someones calls SettingsHeader("ScoobyDoo")

Reply:
In this case it is not an ActivityType but a SettingType, but you are right I created an enum for this so that it is cleaner !